File: pe-core/src/main/java/org/saar/core/renderer/RenderContext.java
package org.saar.core.renderer;

import org.saar.core.camera.ICamera;
import org.saar.lwjgl.opengl.clipplane.ClipPlane;

public class RenderContext {

    private final ICamera camera;
    private final ClipPlane clipPlane;

    public RenderContext(ICamera camera) {
        this(camera, null);
    }

    public RenderContext(ICamera camera, ClipPlane clipPlane) {
        this.camera = camera;
        this.clipPlane = clipPlane;
    }

    public ICamera getCamera() {
        return camera;
    }

    public ClipPlane getClipPlane() {
        return clipPlane;
    }
}

File: pe-core/src/main/java/org/saar/core/renderer/RenderContext.kt
package org.saar.core.renderer;

import org.saar.core.camera.ICamera;

public class RenderContext {

    private final ICamera camera;

    public RenderContext() {
        this.camera = null;
    }

    public RenderContext(ICamera camera) {
        this.camera = camera;
    }

    public RenderContext(RenderContext context) {
        this.camera = context.camera;
    }

    public ICamera getCamera() {
        return camera;
    }
}
